core/src/test/java/org/elasticsearch/bootstrap/BootstrapChecksTests.java

```diff
@@ -560,12 +560,48 @@ private void runMightForkTest(
         consumer.accept(e);
     }
 
+    public void testEarlyAccessCheck() throws NodeValidationException {
+        final AtomicReference<String> javaVersion
+                = new AtomicReference<>(randomFrom("1.8.0_152-ea", "9-ea"));
```

Comment:
Because I set it to a different value further down in the method to simulate a check where the Java version is not an early-access build.